notebooks/cugraph_benchmarks/release.ipynb

```diff
@@ -0,0 +1,628 @@
+    "    t2 = time.time() - t1\n",
+    "    return t2\n",
+    "\n",
+    "def cu_bc(_df):\n",
+    "    t1 = time.time()\n",
+    "    _G = create_cu_ugraph(_df)\n",
+    "    _ = cugraph.betweenness_centrality(_G, k=100)\n",
+    "    t2 = time.time() - t1\n",
+    "    return t2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Louvain"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def nx_louvain(_df):\n",
+    "    t1 = time.time()\n",
+    "    _G = create_nx_ugraph(_df)\n",
+    "    parts = community.best_partition(_G)\n",
+    "    \n",
+    "    # Calculating modularity scores for comparison \n",
+    "    _ = community.modularity(parts, _G)  \n",
+    "    \n",
+    "    t2 = time.time() - t1\n",
+    "    return t2\n",
+    "\n",
+    "def cu_louvain(_df):\n",
+    "    t1 = time.time()\n",
+    "    _G = create_cu_ugraph(_df)\n",
+    "    _,_ = cugraph.louvain(_G)\n",
+    "    t2 = time.time() - t1\n",
+    "    return t2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Triangle Counting"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def nx_tc(_df):\n",
+    "    t1 = time.time()\n",
+    "    _G = create_nx_ugraph(_df)\n",
+    "    nx_count = nx.triangles(_G)\n",
+    "    \n",
+    "    # To get the number of triangles, we would need to loop through the array and add up each count\n",
+    "    count = 0\n",
+    "    for key, value in nx_count.items():\n",
+    "        count = count + value    \n",
+    "    \n",
+    "    t2 = time.time() - t1\n",
+    "    return t2\n",
+    "\n",
+    "def cu_tc(_df):\n",
+    "    t1 = time.time()\n",
+    "    _G = create_cu_ugraph(_df)\n",
+    "    _ = cugraph.triangles(_G)\n",
+    "    t2 = time.time() - t1\n",
+    "    return t2"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Benchmark Functions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# number of datasets\n",
+    "num_datasets = len(data)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# do a simple pass just to get all the libraries initiallized\n",
+    "v = './data/preferentialAttachment.mtx'\n",
+    "gdf = read_data(v)\n",
+    "print(f\"\\tGDF Size {len(gdf)}\")\n",
+    "\n",
+    "g = create_cu_ugraph(gdf)\n",
+    "\n",
+    "print(f\"\\tcugraph Size {g.number_of_edges()}\")\n",
+    "print(f\"\\tcugraph Order {g.number_of_vertices()}\")\n",
+    "\n",
+    "del gdf\n",
+    "del g\n",
+    "gc.collect()"
```

Comment:
We should look into if we can remove all the `del` and `gc.collect()` calls, since they look very suspicious to a new reader (ie. "will I need to do this in my code?", "are there memory management issues with this library?", etc.)  I suspect we can remove them.

Reply:
added additional text